llvm/lib/Target/X86/X86ISelLowering.cpp

```diff
@@ -57581,6 +57581,14 @@ X86TargetLowering::getConstraintType(StringRef Constraint) const {
       case '2':
         return C_RegisterClass;
       }
+    case 'j':
+      switch (Constraint[1]) {
+      default:
+        break;
+      case 'r':
+      case 'R':
+        return C_RegisterClass;
```

Comment:
What's this used for? I can't find `r` in the switch-cases before this PR.

Reply:
`r` is handled at TargetLowering::getConstraintType

Reply:
relate assertion:
llvm/lib/CodeGen/SelectionDAG/SelectionDAGBuilder.cpp:9938: void llvm::SelectionDAGBuilder::visitInlineAsm(const llvm::CallBase&, const llvm::BasicBlock*): Assertion `(OpIn
fo.ConstraintType == TargetLowering::C_RegisterClass || OpInfo.ConstraintType == TargetLowering::C_Register) && "Unknown constraint type!"' failed.